atomic_reactor/plugins/pre_koji_parent.py

```diff
@@ -216,22 +220,35 @@ def manifest_list_entries_match(self, image, build_id):
                 return False
 
             arch = manifest['platform']['architecture']
+            if arch not in self.architectures:
+                self.log.warning('Requested platform %s is missing in manifest list', arch)
+                return False
             v2_digest = manifest['digest']
             manifest_list_data[arch] = v2_digest
 
         archives = self.koji_session.listArchives(build_id)
         koji_archives_data = {}
         for archive in (a for a in archives if a['btype'] == KOJI_BTYPE_IMAGE):
             arch = archive['extra']['docker']['config']['architecture']
+            if arch not in self.architectures:
+                self.log.warning('Requested platform %s is missing in Koji archive', arch)
+                return False
```

Comment:
we should first find all missing arches in koji archives and then report all of them

Reply:
So how about this?
```
        if not all(arch in manifest_list_data for arch in self.architectures):
            self.log.warning('Requested platforms "%s" are missing in manifest list "%s"',
                             self.architectures, manifest_list_data)
            return False
````